lib/Twig/Template.php

```diff
@@ -126,6 +126,34 @@ public function displayParentBlock($name, array $context, array $blocks = array(
     }
 
     /**
+     * Checks if a block exists.
+     *
+     * This method is for internal use only and should never be called
+     * directly.
+     *
+     * @param string $name      The block name to test
+     * @param array  $context   The context
+     * @param array  $blocks    The current set of blocks
+     * @param bool   $useBlocks Whether to use the current set of blocks
+     *
+     * @return bool
+     */
+    public function existsBlock($name, array $context, array $blocks = array(), $useBlocks = true)
+    {
+        $name = (string) $name;
+
+        if ($useBlocks && isset($blocks[$name])) {
+            return $blocks[$name][0] instanceof self;
+        } elseif (isset($this->blocks[$name])) {
+            return $this->blocks[$name][0] instanceof self;
```

Comment:
this can simply be ``return true``. We control exactly what goes in ``$this->blocks``, as this is only coming from Twig code (we need the validation for the ``$blocks`` array as this is the input).